pkg/imgpkg/registry/registry.go

```diff
@@ -203,6 +209,14 @@ func NewSimpleRegistryWithTransport(opts Opts, rTripper http.RoundTripper, regOp
 // the image provided
```

Comment:
Seems like the godoc is slightly out-of-date, now.

In the case where there's no keychain configured, we don't replace it at all.

Does this work?

> // CloneWithSingleAuth produces a copy of this Registry whose keychain has exactly one auth — the one that can be used to access `imageRef`. If no keychain is explicitly configured on this Registry, the copy is a BasicRegistry.